util/src/network/discovery.rs

```diff
@@ -407,25 +407,37 @@ impl Discovery {
 		let target: NodeId = try!(rlp.val_at(0));
 		let timestamp: u64 = try!(rlp.val_at(1));
 		try!(self.check_timestamp(timestamp));
-		let limit = (MAX_DATAGRAM_SIZE - 109) / 90;
 		let nearest = Discovery::nearest_node_entries(&target, &self.node_buckets);
 		if nearest.is_empty() {
 			return Ok(None);
 		}
+		let mut packets = Discovery::prepare_neighbours_packets(&nearest);
+		for p in packets.drain(..) {
+			self.send_packet(PACKET_NEIGHBOURS, &from, &p);
+		}
+		trace!(target: "discovery", "Sent {} Neighbours to {:?}", nearest.len(), &from);
+		Ok(None)
+	}
+
+	fn prepare_neighbours_packets(nearest: &[NodeEntry]) -> Vec<Bytes> {
+		let mut packets = Vec::new();
 		let mut rlp = RlpStream::new_list(1);
-		rlp.begin_list(cmp::min(limit, nearest.len()));
+		let limit = (MAX_DATAGRAM_SIZE - 109) / 90;
+		let mut count = cmp::min(limit, nearest.len()); 
+		rlp.begin_list(count);
 		for n in 0 .. nearest.len() {
 			rlp.begin_list(4);
 			nearest[n].endpoint.to_rlp(&mut rlp);
 			rlp.append(&nearest[n].id);
-			if (n + 1) % limit == 0 || n == nearest.len() - 1 {
-				self.send_packet(PACKET_NEIGHBOURS, &from, &rlp.drain());
-				trace!(target: "discovery", "Sent {} Neighbours to {:?}", n, &from);
+			count -= 1;
+			if count == 0 {
```

Comment:
What do you think about moving this to the beginning of the loop and start with `count=0 && isFirst=false` flags to simplify logic with first/subsequent rlps?